src/ToDoList.js

```diff
@@ -0,0 +1,150 @@
+class ToDoList {
+  constructor() {
+    this.tasksList = [];
+  }
+
+  getTasksListFromStore = () => {
+    const tasks = localStorage.getItem('tasks-list');
+    if (tasks) {
+      this.tasksList = [...JSON.parse(tasks)];
+    }
+  };
+
+  setTasksListToStore = () => {
+    localStorage.setItem('tasks-list', JSON.stringify(this.tasksList));
+  };
+
+  setTasksIds = () => {
+    const tasks = this.tasksList.map((task, i) => ({
+      description: task.description,
+      completed: task.completed,
+      index: i + 1,
+    }));
+    this.tasksList = [...tasks];
+  };
+
+  markUpTask = (task) => `
+    <li class="task ${task.completed ? 'completed' : ''}" data-index="${task.index}">
+      <div class="task-content">
+        <div class="mark-task">
+          <input type="checkbox" id="task-${task.index}-status" ${task.completed ? 'checked' : ''}>
+          <label for="task-${task.index}-status"></label>
+        </div>
+        <h3 class="task-description" contenteditable="true">${
+          task.description
+        }</h3>
+      </div>
+      <div class="task-options">
+        <div class="drag-icon">
+          <span class="dot"></span>
+          <span class="dot"></span>
+          <span class="dot"></span>
+        </div>
+      </div>
+      <button type="button" class="remove-task" disabled>
+        <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="#bebebe" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><polyline points="3 6 5 6 21 6"></polyline><path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path><line x1="10" y1="11" x2="10" y2="17"></line><line x1="14" y1="11" x2="14" y2="17"></line></svg>
+      </button>
+    </li>
+  `;
+
+  renderTasksList = () => {
+    const listItems = this.tasksList.map((task) => this.markUpTask(task));
+    document.querySelector('#to-do-list').innerHTML = listItems.join('');
+    this.generalEventHandlers();
+  };
+
+  createTask = (inputDesc) => {
+    const inputDescription = inputDesc.value;
+    if (inputDescription.length > 0) {
+      const id = this.tasksList.length + 1;
+      const newTask = {
+        description: inputDescription,
+        completed: false,
+        index: id,
+      };
+      this.tasksList.push(newTask);
+      inputDesc.value = '';
+      inputDesc.focus();
+      this.setTasksListToStore();
+      this.renderTasksList();
+    }
+  };
+
+  updateTask = (event) => {
+    const taskItem = event.target;
+    const taskId = taskItem.parentElement.parentElement.dataset.index;
+    const index = Number(taskId) - 1;
+    const taskDescription = taskItem.innerText;
+    this.tasksList[index].description = taskDescription;
+  };
+
+  deleteTask = (event) => {
```

Comment:
* Kindly look into this bug as i couldn't delete any of the todos i created below is a recording of the bug 👇 

https://github.com/NitBravoA92/project-to-do-list/assets/52497246/881c21b9-fe97-4495-92cd-66acbc1f0c74

Reply:
Hi, @Davidosky007  How are you? Thank you very much for taking the time to review my code and leaving me your valuable review. After reviewing my code and making some small tweaks, I'll leave you a video demo of the app running on 3 different browsers: Firefox Browser, Google Chrome, and Microsoft Edge. I also tested the functionality using the final files in the dist folder with the same results as in the video. I would really appreciate it if you would let me know if you are still experiencing the same issue above with the delete functionality.

Demo: https://www.loom.com/share/fc52acc581a545e48f557a4ccb356d7d